lib/ansible/modules/packaging/os/redhat_repository.py

```diff
@@ -0,0 +1,260 @@
+        if repo_id_re:
+            repo_id = repo_id_re.group(1)
+            continue
+
+        repo_name_re = re.match(repo_name_re_str, line)
+        if repo_name_re:
+            repo_name = repo_name_re.group(1)
+            continue
+
+        repo_url_re = re.match(repo_url_re_str, line)
+        if repo_url_re:
+            repo_url = repo_url_re.group(1)
+            continue
+
+        repo_enabled_re = re.match(repo_enabled_re_str, line)
+        if repo_enabled_re:
+            repo_enabled = repo_enabled_re.group(1)
+
+            repo = {"id": repo_id, "name": repo_name, "url": repo_url, "enabled": True if repo_enabled == '1' else False}
+            repo_result.append(repo)
+    return repo_result
+
+
+def repository_list(module, list_parameter):
+    # Get rhsm repository list and format it for the user
+    repo = get_repository_list(module, list_parameter)
+    module.exit_json(changed=False, repositories=repo)
+
+
+def repository_modify(module, state, name):
+    current_repo_list = get_repository_list(module, 'all')
+    name = set(name)
+
+    matched_existing_repo = {}
+    for repoid in name:
+        matched_existing_repo[repoid] = []
+        for repo in current_repo_list:
+            if fnmatch(repo['id'], repoid):
+                matched_existing_repo[repoid].append(repo)
+
+    changed = False
+    results = []
+    rhsm_arguments = ['repos']
+
+    for repoid in matched_existing_repo:
+        if len(matched_existing_repo[repoid]) == 0:
+            results.append("%s is not a valid repository ID" % repoid)
+            module.fail_json(results=results, msg="%s is not a valid repository ID" % repoid)
+        for repo in matched_existing_repo[repoid]:
+            if state == 'disabled':
+                if repo['enabled']:
+                    changed = True
+                if module.check_mode:
+                    results.append("Repository '%s' is disabled for this system" % repo['id'])
+                else:
+                    rhsm_arguments += ['--disable', repoid]
+            elif state == 'enabled':
+                if not repo['enabled']:
+                    changed = True
+                if module.check_mode:
+                    results.append("Repository '%s' is enabled for this system" % repo['id'])
+                else:
+                    rhsm_arguments += ['--enable', repoid]
+
+    if module.check_mode:
+        module.exit_json(results=results, changed=changed)
+    rc, out, err = run_subscription_manager(module, rhsm_arguments)
+    results = out.split('\n')
+    module.exit_json(results=results)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            list=dict(choices=['all', 'enabled', 'disabled'],
+                      required=False),
+            name=dict(type="list",
+                      required=False),
+            state=dict(choices=['enabled', 'disabled'],
+                       default=None,
```

Comment:
Also the convention is to be `present` and `absent` the difference in the `yum_repository` module is that
`present|absent` means the repo is installed and `enabled|disabled` means the repo is in use. With yum repos you can have a present repository but in a disabled state meaning the .repo file is there but there is a flag in it telling yum not to use it.